extract/eventlog/eventlog.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package eventlog provides utilities for interpreting Canonical Event Log events.
+package eventlog
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"unicode/utf16"
+
+	"github.com/google/gce-tcb-verifier/eventlog"
+	oabi "github.com/google/gce-tcb-verifier/ovmf/abi"
+	"github.com/google/go-sev-guest/verify/trust"
+)
+
+// ErrLocateGetterNil is returned when a LocateOptions.Getter is nil.
+var ErrLocateGetterNil = errors.New("locate getter is nil")
+
+// LocateOptions contains options for locating data that can be local or remote.
+type LocateOptions struct {
+	Getter               trust.HTTPSGetter
+	EfiVarsMountLocation string
+}
+
+// DefaultLocateOptions returns a default LocateOptions.
+func DefaultLocateOptions() *LocateOptions {
+	return &LocateOptions{
+		Getter:               trust.DefaultHTTPSGetter(),
+		EfiVarsMountLocation: "/sys/firmware/efi/efivars",
+	}
+}
+
+// variableLocatorPath reads a UEFI variable name from a RIM locator of variable type.
+func variableLocatorPath(dst *string, data []byte) error {
```

Comment:
why does this return via out parameter instead of returning (string, error) which I thought was more standard go style?